### ISSUE TITLE
'make' fails if lzma.h not found, and even after 'lzma' package is installed

### ISSUE THREAD

**Issue by jkeenan** [pmqs/Compress-Raw-Lzma#7]
Today I attempted to install Compress::Raw::Lzma on Linux (Ubuntu 22.04 LTS) against perl-5.38.0.  I was not previously familiar with this distribution, but I noticed that you issued a new release at the same time as the new Compress::Raw::Zlib and Compress::Raw::Bzip2.

My installation failed, apparently due to lack of an `lzma.h` header file.  (I had no problem installing it on FreeBSD-12.)
```
$ uname -mrs
Linux 5.19.0-46-generic x86_64

$ perl -v | head -2 | tail -1
This is perl 5, version 38, subversion 0 (v5.38.0) built for x86_64-linux

$ cpan Compress::Raw::Lzma
Reading '/home/jkeenan/.cpan/Metadata'
  Database was generated on Tue, 18 Jul 2023 00:17:01 GMT
Running install for module 'Compress::Raw::Lzma'
CPAN: Digest::SHA loaded ok (v6.04)
CPAN: Compress::Zlib loaded ok (v2.204)
Checksum for /home/jkeenan/.cpan/sources/authors/id/P/PM/PMQS/Compress-Raw-Lzma-2.205.tar.gz ok
'YAML' not installed, will not store persistent state
CPAN: CPAN::Meta::Requirements loaded ok (v2.140)
CPAN: Parse::CPAN::Meta loaded ok (v2.150010)
CPAN: CPAN::Meta loaded ok (v2.150010)
CPAN: Module::CoreList loaded ok (v5.20230520)
Configuring P/PM/PMQS/Compress-Raw-Lzma-2.205.tar.gz with Makefile.PL
CPAN: CPAN::Reporter loaded ok (v1.2019)
Parsing config.in...
Looks Good.
Up/Downgrade not needed.
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Compress::Raw::Lzma
Writing MYMETA.yml and MYMETA.json
(/home/jkeenan/perl5/perlbrew/perls/perl-5.38.0/bin/perl Makefile.PL exited with 0)
CPAN::Reporter: Makefile.PL result is 'pass', No errors.
  PMQS/Compress-Raw-Lzma-2.205.tar.gz
  /home/jkeenan/perl5/perlbrew/perls/perl-5.38.0/bin/perl Makefile.PL -- OK
Running make for P/PM/PMQS/Compress-Raw-Lzma-2.205.tar.gz
cp lib/Compress/Raw/Lzma.pm blib/lib/Compress/Raw/Lzma.pm
AutoSplitting blib/lib/Compress/Raw/Lzma.pm (blib/lib/auto/Compress/Raw/Lzma)
Running Mkbootstrap for Lzma ()
chmod 644 "Lzma.bs"
"/home/jkeenan/perl5/perlbrew/perls/perl-5.38.0/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Lzma.bs blib/arch/auto/Compress/Raw/Lzma/Lzma.bs 644
"/home/jkeenan/perl5/perlbrew/perls/perl-5.38.0/bin/perl" "/home/jkeenan/perl5/perlbrew/perls/perl-5.38.0/lib/5.38.0/ExtUtils/xsubpp"  -typemap '/home/jkeenan/perl5/perlbrew/perls/perl-5.38.0/lib/5.38.0/ExtUtils/typemap' -typemap '/home/jkeenan/.cpan/build/Compress-Raw-Lzma-2.205-1/typemap'  Lzma.xs > Lzma.xsc
mv Lzma.xsc Lzma.c
cc -c  -I/usr/local/include -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2   -DVERSION=\"2.205\" -DXS_VERSION=\"2.205\" -fPIC "-I/home/jkeenan/perl5/perlbrew/perls/perl-5.38.0/lib/5.38.0/x86_64-linux/CORE"   Lzma.c
Lzma.xs:18:10: fatal error: lzma.h: No such file or directory
   18 | #include "lzma.h"
      |          ^~~~~~~~
compilation terminated.
make: *** [Makefile:347: Lzma.o] Error 1
(/usr/bin/make exited with 512)
CPAN::Reporter: make result is 'unknown', Stopped with an error.
CPAN::Reporter: preparing a CPAN Testers report for Compress-Raw-Lzma-2.205

CPAN::Reporter: this appears to be a duplicate report for the make phase:
UNKNOWN Compress-Raw-Lzma-2.205 x86_64-linux 5.19.0-45-generic

Test report will not be sent.

  PMQS/Compress-Raw-Lzma-2.205.tar.gz
  /usr/bin/make -- NOT OK
```
Would it be possible to have `make` exit with a more specific error message if this header is not found?

Note also:  I installed the `lzma` package:
```
$ sudo apt install lzma
```
... but `make` still failed:
```
...
cc -c  -I/usr/local/include -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2   -DVERSION=\"2.205\" -DXS_VERSION=\"2.205\" -fPIC "-I/home/jkeenan/perl5/perlbrew/perls/perl-5.38.0/lib/5.38.0/x86_64-linux/CORE"   Lzma.c
Lzma.xs:18:10: fatal error: lzma.h: No such file or directory
   18 | #include "lzma.h"
      |          ^~~~~~~~
compilation terminated.
make: *** [Makefile:347: Lzma.o] Error 1
(/usr/bin/make exited with 512)
```
Suggestions?

**Comment by jkeenan** [comment]
My next idea was to install the `lzma-dev` package.  I did install that package, but got no better results with the `make` stage of installing this distribution.

**Comment by pmqs** [comment]
Very strange. My dev box is one of the Ubuntu variants, so should be the same as you. 


The `lzma.h` file should be under `/usr/include`

```
$ locate lzma.h
/usr/include/lzma.h
/usr/include/boost/iostreams/filter/lzma.hpp
/usr/share/doc/liblzma-dev/html/structlzma__options__lzma.html
/usr/src/linux-headers-6.2.0-23/include/linux/decompress/unlzma.h
/usr/src/linux-headers-6.2.0-25/include/linux/decompress/unlzma.h
```


that files comes from `liblzma-dev`
```
$ apt-file search /usr/include/lzma.h
liblzma-dev: /usr/include/lzma.h          
```

I see you have installed `lzma-dev` rather than `liblzma-dev`. That doesn't appear to have `lzma.h`

```
$ apt-file list lzma-dev | grep lzma.h
```
What is the different  with these two package?
```
$ apt-cache show lzma-dev | grep Homepage
Homepage: http://www.7-zip.org/sdk.html

$ apt-cache show liblzma-dev | grep Homepage
Homepage: https://tukaani.org/xz/
```
So `lzma-dev` is sourced from 7-zip, while `liblzma-dev` comes from the xz site (which is the one I need). Suspect the z-zip package got squatters rights on the `lzma-dev` name, so xz had to use something else.

Can you give `liblzma-dev`  a try?

If that works I'll  add a note in the README.

**Comment by pmqs** [comment]
Excellent. Fix b2d20bb04a940f29e743b83396253794cde33a2d push to GH to update README
